### ISSUE TITLE
windows10 - `nmake install` fails with fatal error u1077: "pod2html": return-code "0xff"

### ISSUE THREAD

**Issue by ML0401** [openssl/openssl#10648]
I am trying to install openssl on windows 10, therefor i have already installed and downloaded:

- Active State Perl 5.26.3 (MSWin32_x64)
- visual studio 17 and 19 (only one required)
- openssl-master folder

and executed the following steps via admin and command prompt:
> cd \my path to Perl folder\
> perl \my path to Configure file from openssl-master\ no-asm VC-WIN32
> nmake (no errors)
> nmake test (test pass, no errors)
> nmake install gets me the following error message:

The command "href" is wrong or could not be found 
NMAKE: fatal error u1077: "pod2html": return-code "0xff"

I have already installed openssl the same way on windows7 without any complications. Am I missing something or mixing up the different version? VC-WIN64A and VC-WIN64I are not working as well. Thanks in advance for any kind of help.

**Comment by mattcaswell** [comment]
> NMAKE: fatal error u1077: "opd2html": return-code "0xff"

Is this a typo? Does this actually say "pod2html", or is this the real error?

It seems that it is failing for some reason during installation of the documentation. If you don't care about that a workaround might be to only install the software itself, i.e.

````
nmake install_sw
````

**Comment by ML0401** [comment]
Yes, you are right. I changed it to “pod2html“ and will check if it is working. Thank you!

**Comment by ML0401** [comment]
> > NMAKE: fatal error u1077: "opd2html": return-code "0xff"
> 
> Is this a typo? Does this actually say "pod2html", or is this the real error?
> 
> It seems that it is failing for some reason during installation of the documentation. If you don't care about that a workaround might be to only install the software itself, i.e.
> 
> ```
> nmake install_sw
> ```

**Comment by ML0401** [comment]
I changed it. You were right it is “pod2html”. I will try your solution. I need OpenSSL to create certificates, so I hope this will work.

**Comment by mspncp** [comment]
I can confirm this bug. I just encountered it on my notebook which is also running Windows 10.

```
...
*** Installing runtime programs
Copying: apps//openssl.exe to /dev/openssl/bin/openssl.exe
Copying: apps//openssl.pdb to /dev/openssl/bin/openssl.pdb
Copying: tools//c_rehash.pl to /dev/openssl/bin/c_rehash.pl
Copying: ./apps/openssl.cnf to /dev/openssl/ssl/openssl.cnf.dist
Copying: apps//CA.pl to /dev/openssl/ssl/misc/CA.pl
Copying: apps//tsget.pl to /dev/openssl/ssl/misc/tsget.pl
Copying: ./apps/ct_log_list.cnf to /dev/openssl/ssl/ct_log_list.cnf.dist
NMAKE : warning U4006: Spezialmakro undefiniert: "$<""
        pod2html "--podroot=./doc" --htmldir=..  --podpath=man1:man3:man5:man7 "--infile=" "--title=CA.pl"  | C:\Perl64\bin\perl.exe -pe "s|href=\"http://man\.he\.net/(mand/[^\"]+)(?:\.html)?\"|href=\"../.html|g;"  > doc\html\man1\CA.pl.html
Der Befehl "href" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.
NMAKE : fatal error U1077: "pod2html": Rückgabe-Code "0xff"
Stop.
```

**Comment by mspncp** [comment]
I marked this issue urgent, because the failure of `nmake install` breaks the extended tests on AppVeyor, see e.g. job [master.30343](https://ci.appveyor.com/project/openssl/openssl/builds/29781011/job/4kavu0icpa0inmc7). The reason is that for the extended tests a temporary installation is attempted:

```
cmd /c "nmake install DESTDIR=..\_install 2>&1"
```

**Comment by mspncp** [comment]
It's actually `nmake install_html_docs` which fails.

**Comment by mspncp** [comment]
@levitte it seems like nmake is having problems with the double extensions `.pl.html <- .pl.pod` in [inference rules](https://docs.microsoft.com/en-us/cpp/build/reference/inference-rules?view=vs-2019).

**Comment by mspncp** [comment]
See also the [nmake documentation for $<](https://docs.microsoft.com/en-us/cpp/build/reference/filename-macros?redirectedfrom=MSDN&view=vs-2019):


    $< | Dependent file with a later timestamp than the current target.
         Valid only in commands in inference rules.

**Comment by mspncp** [comment]
> I changed it. You were right it is “pod2html”. I will try your solution. I need OpenSSL to create certificates, so I hope this will work.

@ML0401 if you only need OpenSSL to create certificates, you might be better of using the most current version of the **OpenSSL_1_1_1-stable** branch instead of the **master** branch, which is under development. You can either use git to fetch and checkout that branch or download the latest release [openssl-1.1.1d.tar.gz](https://www.openssl.org/source/openssl-1.1.1d.tar.gz) from our webpage.

**Comment by levitte** [comment]
[swears]
I thought all make implementations always defined `$<`, even when not in an implicit rule (which we *don't* use). Of course, Microsoft had to do things differently (and less compatible).

I'll fix when I'm back from vacation.

**Comment by mspncp** [comment]
> I'll fix when I'm back from vacation.

Enjoy your vacation. The 'urgent' label only applies to the 24 hour rule ;-)

**Comment by mspncp** [comment]
For a fix, see #10719.
